vip-api/src/main/java/fr/insalyon/creatis/vip/api/controller/admin/AppVersionController.java

```diff
@@ -0,0 +1,176 @@
+            }
+            return appVersions;
+        } catch (BusinessException e) {
+            throw new ApiException(e);
+        }
+    }
+
+    private class AppVersionStrings {
+        public String appName;
+        public String version;
+        AppVersionStrings(String appName, String version) {
+            this.appName = appName;
+            this.version = version;
+        }
+    }
+
+    private AppVersionStrings parseAppVersionId(String appVersionId) throws ApiException {
+        try {
+            int delimiterPos = appVersionId.lastIndexOf("/");
+            if (delimiterPos >= 0) {
+                String appName = appVersionId.substring(0, delimiterPos);
+                String version = appVersionId.substring(delimiterPos + 1);
+                return new AppVersionStrings(appName, version);
+            } else {
+                throw new ApiException("Invalid appVersionId");
+            }
+        } catch (ApiException e) {
+            logger.error("Error decoding appVersionId {}", appVersionId, e);
+            throw new ApiException("cannot decode appVersionId : " + appVersionId);
+        }
+
+    }
+
+    @RequestMapping(value = "{appVersionId}", method = RequestMethod.GET)
+    public AppVersion getAppVersion(@PathVariable String appVersionId) throws ApiException {
+        logMethodInvocation(logger, "getAppVersion", appVersionId);
+        AppVersionStrings input = parseAppVersionId(appVersionId);
+        try {
+            AppVersion appVersion = appVersionBusiness.getVersion(input.appName, input.version);
+            if (appVersion == null) {
+                throw new ApiException("Not found");
+            }
+            return appVersion;
+        } catch (BusinessException e) {
+            throw new ApiException(e);
+        }
+    }
+
+    @RequestMapping(value = "{appVersionIdFirstPart}/{appVersionIdSecondPart}", method = RequestMethod.GET)
+    public AppVersion getAppVersion(@PathVariable String appVersionIdFirstPart,
+                                    @PathVariable String appVersionIdSecondPart) throws ApiException {
+        return getAppVersion(appVersionIdFirstPart + "/" + appVersionIdSecondPart);
+    }
+
+    @RequestMapping(value = "/{appVersionId}", method = RequestMethod.PUT)
+    public AppVersion createOrUpdateAppVersion(@PathVariable String appVersionId,
+                                               @RequestBody @Valid AppVersion appVersion) throws ApiException {
+        logMethodInvocation(logger, "createOrUpdateAppVersion", appVersionId);
+        AppVersionStrings input = parseAppVersionId(appVersionId);
+        try {
+            AppVersion existingAppVersion = appVersionBusiness.getVersion(input.appName, input.version);
+            if (existingAppVersion == null) {
+                appVersionBusiness.add(appVersion);
+            } else {
+                appVersionBusiness.update(appVersion);
```

Comment:
don't we verify the name and version in the AppVersion object correspond to the one on the URL ?